casper/src/main/scala/coop/rchain/casper/BlockCreator.scala

```diff
@@ -65,19 +71,33 @@ object BlockCreator {
     spanF.trace(CreateBlockMetricsSource) {
       import cats.instances.list._
       for {
-        tipHashes             <- Estimator[F].tips(dag, genesis)
-        _                     <- spanF.mark("after-estimator")
-        parentMetadatas       <- EstimatorHelper.chooseNonConflicting(tipHashes, dag)
+        tipHashes <- Estimator[F].tips(dag, genesis)
+        _ <- Log[F].info(
+              s"Creating block with tip hashes ${tipHashes.map(PrettyPrinter.buildString)}"
+            )
+        _               <- spanF.mark("after-estimator")
+        parentMetadatas <- EstimatorHelper.chooseNonConflicting(tipHashes, dag)
+        _ <- Log[F].info(
+              s"Creating block with parents ${parentMetadatas.map(b => PrettyPrinter.buildString(b.blockHash))}"
+            )
         maxBlockNumber        = ProtoUtil.maxBlockNumberMetadata(parentMetadatas)
+        _                     <- Log[F].info(s"Creating block with maxBlockNumber ${maxBlockNumber}")
         invalidLatestMessages <- ProtoUtil.invalidLatestMessages(dag)
         deploys               <- extractDeploys(dag, parentMetadatas, maxBlockNumber, expirationThreshold)
         sender                = ByteString.copyFrom(validatorIdentity.publicKey.bytes)
         latestMessageOpt      <- dag.latestMessage(sender)
         seqNum                = latestMessageOpt.fold(0)(_.seqNum) + 1
+        _                     <- Log[F].info(s"Creating block with seqNum ${seqNum}")
+        // if the node is already not bonded by the parent, the node won't slash once more
+        invalidLatestMessagesExcludeUnbonded = invalidLatestMessages.filter(
+          v =>
+            parentMetadatas
+              .map(p => isBonded(p, v._1))
+              .forall(identity)
```

Comment:
Only the first (base) parent needs to be checked, this is the same as getting bonds list from base parent block.